### ISSUE TITLE
Error compiling optional_ops.cc when building from source with MSVC 2017

### ISSUE THREAD

**Issue by lakeofsoft** [tensorflow/tensorflow#22387]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: NO
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Microsoft Windows [Version 10.0.17134.228]
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: N/A
- **TensorFlow installed from (source or binary)**: Source
- **TensorFlow version (use command below)**: trunk
- **Python version**: Python 3.7.0
- **Bazel version (if compiling from source)**: Build label: 0.17.1
- **GCC/Compiler version (if compiling from source)**: Microsoft (R) C/C++ Optimizing Compiler Version 19.15.26729 for x64
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: 
bazel build --config=opt //tensorflow/tools/pip_package:build_pip_package

### Describe the problem
Unable to build from sources with Microsoft Build Tools 2017. Error when compiling **tensorflow/core/kernels/data/optional_ops.cc**

### Source code / logs
```
C:/Program Files (x86)/Microsoft Visual Studio/2017/Professional/VC/Tools/MSVC/14.15.26726/bin/HostX64/x64/cl.exe /nologo /DCOMPILER_MSVC /DNOMINMAX /D_WIN32_WINNT=0x0600 /D_CRT_SECURE_NO_DEPRECATE /D_CRT_SECURE_NO_WARNINGS /bigobj /Zm500 /EHsc /wd4351 /wd4291 /wd4250 /wd4996 /I. /Ibazel-out/x64_windows-opt/genfiles /Ibazel-out/x64_windows-opt/bin /Iexternal/bazel_tools /Ibazel-out/x64_windows-opt/genfiles/external/bazel_tools /Ibazel-out/x64_windows-opt/bin/external/bazel_tools /Iexternal/eigen_archive /Ibazel-out/x64_windows-opt/genfiles/external/eigen_archive /Ibazel-out/x64_windows-opt/bin/external/eigen_archive /Iexternal/local_config_sycl /Ibazel-out/x64_windows-opt/genfiles/external/local_config_sycl /Ibazel-out/x64_windows-opt/bin/external/local_config_sycl /Iexternal/com_google_absl /Ibazel-out/x64_windows-opt/genfiles/external/com_google_absl /Ibazel-out/x64_windows-opt/bin/external/com_google_absl /Iexternal/nsync /Ibazel-out/x64_windows-opt/genfiles/external/nsync /Ibazel-out/x64_windows-opt/bin/external/nsync /Iexternal/gif_archive /Ibazel-out/x64_windows-opt/genfiles/external/gif_archive /Ibazel-out/x64_windows-opt/bin/external/gif_archive /Iexternal/jpeg /Ibazel-out/x64_windows-opt/genfiles/external/jpeg /Ibazel-out/x64_windows-opt/bin/external/jpeg /Iexternal/protobuf_archive /Ibazel-out/x64_windows-opt/genfiles/external/protobuf_archive /Ibazel-out/x64_windows-opt/bin/external/protobuf_archive /Iexternal/com_googlesource_code_re2 /Ibazel-out/x64_windows-opt/genfiles/external/com_googlesource_code_re2 /Ibazel-out/x64_windows-opt/bin/external/com_googlesource_code_re2 /Iexternal/farmhash_archive /Ibazel-out/x64_windows-opt/genfiles/external/farmhash_archive /Ibazel-out/x64_windows-opt/bin/external/farmhash_archive /Iexternal/fft2d /Ibazel-out/x64_windows-opt/genfiles/external/fft2d /Ibazel-out/x64_windows-opt/bin/external/fft2d /Iexternal/highwayhash /Ibazel-out/x64_windows-opt/genfiles/external/highwayhash /Ibazel-out/x64_windows-opt/bin/external/highwayhash /Iexternal/zlib_archive /Ibazel-out/x64_windows-opt/genfiles/external/zlib_archive /Ibazel-out/x64_windows-opt/bin/external/zlib_archive /Iexternal/double_conversion /Ibazel-out/x64_windows-opt/genfiles/external/double_conversion /Ibazel-out/x64_windows-opt/bin/external/double_conversion /Iexternal/snappy /Ibazel-out/x64_windows-opt/genfiles/external/snappy /Ibazel-out/x64_windows-opt/bin/external/snappy /Iexternal/com_google_absl /Ibazel-out/x64_windows-opt/genfiles/external/com_google_absl /Ibazel-out/x64_windows-opt/bin/external/com_google_absl /Iexternal/eigen_archive /Ibazel-out/x64_windows-opt/genfiles/external/eigen_archive /Ibazel-out/x64_windows-opt/bin/external/eigen_archive /Iexternal/nsync/public /Ibazel-out/x64_windows-opt/genfiles/external/nsync/public /Ibazel-out/x64_windows-opt/bin/external/nsync/public /Iexternal/gif_archive/lib /Ibazel-out/x64_windows-opt/genfiles/external/gif_archive/lib /Ibazel-out/x64_windows-opt/bin/external/gif_archive/lib /Iexternal/gif_archive/windows /Ibazel-out/x64_windows-opt/genfiles/external/gif_archive/windows /Ibazel-out/x64_windows-opt/bin/external/gif_archive/windows /Iexternal/protobuf_archive/src /Ibazel-out/x64_windows-opt/genfiles/external/protobuf_archive/src /Ibazel-out/x64_windows-opt/bin/external/protobuf_archive/src /Iexternal/farmhash_archive/src /Ibazel-out/x64_windows-opt/genfiles/external/farmhash_archive/src /Ibazel-out/x64_windows-opt/bin/external/farmhash_archive/src /Iexternal/zlib_archive /Ibazel-out/x64_windows-opt/genfiles/external/zlib_archive /Ibazel-out/x64_windows-opt/bin/external/zlib_archive /Iexternal/double_conversion /Ibazel-out/x64_windows-opt/genfiles/external/double_conversion /Ibazel-out/x64_windows-opt/bin/external/double_conversion /DEIGEN_MPL2_ONLY /DEIGEN_MAX_ALIGN_BYTES=64 /D__CLANG_SUPPORT_DYN_ANNOTATION__ /DTF_USE_SNAPPY /showIncludes /MD /O2 /Oy- /DNDEBUG /wd4117 -D__DATE__="redacted" -D__TIMESTAMP__="redacted" -D__TIME__="redacted" /Gy /Gw -w /arch:AVX -DTENSORFLOW_MONOLITHIC_BUILD /DPLATFORM_WINDOWS /DEIGEN_HAS_C99_MATH /DTENSORFLOW_USE_EIGEN_THREADPOOL /DEIGEN_AVOID_STL_ARRAY /Iexternal/gemmlowp /wd4018 /wd4577 /DNOGDI /DTF_COMPILE_LIBRARY /Fobazel-out/x64_windows-opt/bin/tensorflow/core/kernels/data/_objs/optional_ops/optional_ops.obj /c tensorflow/core/kernels/data/optional_ops.cc
C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.15.26726\include\xmemory(217): error C2678: binary '*': no operator found which takes a left-hand operand of type 'const _Iter' (or there is no acceptable conversion)
        with
        [
            _Iter=tensorflow::OpInputList::Iterator
        ]
.\tensorflow/core/framework/op_kernel.h(404): note: could be 'const tensorflow::Tensor &tensorflow::OpArgIterator<tensorflow::OpInputList,const tensorflow::Tensor>::operator *(void)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.15.26726\include\xmemory(217): note: while trying to match the argument list '(const _Iter)'
        with
        [
            _Iter=tensorflow::OpInputList::Iterator
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.15.26726\include\vector(1823): note: see reference to function template instantiation '_FwdIt *std::_Uninitialized_copy<_Iter,tensorflow::Tensor*,std::allocator<_Ty>>(const _InIt,const _InIt,_FwdIt,_Alloc &)' being compiled
        with
        [
            _FwdIt=tensorflow::Tensor *,
            _Iter=tensorflow::OpInputList::Iterator,
            _Ty=tensorflow::Tensor,
            _InIt=tensorflow::OpInputList::Iterator,
            _Alloc=std::allocator<tensorflow::Tensor>
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.15.26726\include\vector(708): note: see reference to function template instantiation 'tensorflow::Tensor *std::vector<tensorflow::Tensor,std::allocator<_Ty>>::_Ucopy<_Iter>(_Iter,_Iter,tensorflow::Tensor *)' being compiled
        with
        [
            _Ty=tensorflow::Tensor,
            _Iter=tensorflow::OpInputList::Iterator
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.15.26726\include\vector(708): note: see reference to function template instantiation 'tensorflow::Tensor *std::vector<tensorflow::Tensor,std::allocator<_Ty>>::_Ucopy<_Iter>(_Iter,_Iter,tensorflow::Tensor *)' being compiled
        with
        [
            _Ty=tensorflow::Tensor,
            _Iter=tensorflow::OpInputList::Iterator
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.15.26726\include\vector(723): note: see reference to function template instantiation 'void std::vector<tensorflow::Tensor,std::allocator<_Ty>>::_Range_construct_or_tidy<_Iter>(_Iter,_Iter,std::forward_iterator_tag)' being compiled
        with
        [
            _Ty=tensorflow::Tensor,
            _Iter=tensorflow::OpInputList::Iterator
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.15.26726\include\vector(723): note: see reference to function template instantiation 'void std::vector<tensorflow::Tensor,std::allocator<_Ty>>::_Range_construct_or_tidy<_Iter>(_Iter,_Iter,std::forward_iterator_tag)' being compiled
        with
        [
            _Ty=tensorflow::Tensor,
            _Iter=tensorflow::OpInputList::Iterator
        ]
tensorflow/core/kernels/data/optional_ops.cc(112): note: see reference to function template instantiation 'std::vector<tensorflow::Tensor,std::allocator<_Ty>>::vector<tensorflow::OpInputList::Iterator,void>(_Iter,_Iter,const _Alloc &)' being compiled
        with
        [
            _Ty=tensorflow::Tensor,
            _Iter=tensorflow::OpInputList::Iterator,
            _Alloc=std::allocator<tensorflow::Tensor>
        ]
tensorflow/core/kernels/data/optional_ops.cc(111): note: see reference to function template instantiation 'std::vector<tensorflow::Tensor,std::allocator<_Ty>>::vector<tensorflow::OpInputList::Iterator,void>(_Iter,_Iter,const _Alloc &)' being compiled
        with
        [
            _Ty=tensorflow::Tensor,
            _Iter=tensorflow::OpInputList::Iterator,
            _Alloc=std::allocator<tensorflow::Tensor>
        ]
external/com_google_absl\absl/container/inlined_vector.h(73): note: see reference to class template instantiation 'std::allocator<T>' being compiled
        with
        [
            T=tensorflow::MemoryType
        ]
.\tensorflow/core/framework/op_kernel.h(176): note: see reference to class template instantiation 'absl::InlinedVector<tensorflow::MemoryType,4,std::allocator<T>>' being compiled
        with
        [
            T=tensorflow::MemoryType
        ]
external/com_google_absl\absl/container/inlined_vector.h(73): note: see reference to class template instantiation 'std::allocator<T>' being compiled
        with
        [
            T=tensorflow::TensorReference
        ]
.\tensorflow/core/framework/unique_tensor_references.h(66): note: see reference to class template instantiation 'absl::InlinedVector<tensorflow::TensorReference,4,std::allocator<T>>' being compiled
        with
        [
            T=tensorflow::TensorReference
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.15.26726\include\xmemory(217): error C2100: illegal indirection
C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.15.26726\include\xmemory(217): error C2062: type 'unknown-type' unexpected
Target //tensorflow/tools/pip_package:build_pip_package failed to build

```

**Comment by lakeofsoft** [comment]
Found the solution ( hinted by #15925 ). In file **/tensorflow/core/framework/op_kernel.h** 
add **const** at operator* for OpArgIterator class:

`reference operator*() const { return (*list_)[i_]; }`

**Comment by tensorflowbutler** [comment]
Nagging Assignee @gunan: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by gunan** [comment]
Looks like this was resolved?
Sorry I missed when it was assigned to me.

**Comment by aaniin** [comment]
Dear @gunan ,
I experienced the same compilation failure as @lakeofsoft , when trying to compile today's tensorflow master with Visual Studio 2017. Adding the `const` as [described above](https://github.com/tensorflow/tensorflow/issues/22387#issuecomment-422925731) let to successful compilation.
Is adding the const appropriate?
**/tensorflow/core/framework/op_kernel.h**
```reference operator*() const { return (*list_)[i_]; }``` 

Build error without the const added.
```ERROR: C:/users/hbeck3/tf-gpu/tensorflow/tensorflow/core/kernels/data/BUILD:270:1: C++ compilation of rule '//tensorflow/core/kernels/data:parse_example_dataset_op' failed (Exit 2): msvc_wrapper_for_nvcc.bat failed: error executing command
  cd C:/users/hbeck3/_bazel_hbeck3/gyijrzyg/execroot/org_tensorflow
  SET CUDA_TOOLKIT_PATH=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.0
    SET CUDNN_INSTALL_PATH=C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.0
    SET INCLUDE=C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.15.26726\ATLMFC\include;C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.15.26726\include;C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt;C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared;C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um;C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt;C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt
    SET LIB=C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.15.26726\ATLMFC\lib\x64;C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.15.26726\lib\x64;C:\Program Files (x86)\Windows Kits\10\lib\10.0.17763.0\ucrt\x64;C:\Program Files (x86)\Windows Kits\10\lib\10.0.17763.0\um\x64;
    SET PATH=C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.15.26726\bin\HostX64\x64;C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\VC\VCPackages;C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\CommonExtensions\Microsoft\TestWindow;C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer;C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\bin\Roslyn;C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Team Tools\Performance Tools\x64;C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Team Tools\Performance Tools;C:\Program Files (x86)\Microsoft Visual Studio\Shared\Common\VSPerfCollectionTools\\x64;C:\Program Files (x86)\Microsoft Visual Studio\Shared\Common\VSPerfCollectionTools\;C:\Program Files (x86)\Windows Kits\10\bin\10.0.17763.0\x64;C:\Program Files (x86)\Windows Kits\10\bin\x64;C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\\MSBuild\15.0\bin;C:\Windows\Microsoft.NET\Framework64\v4.0.30319;C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\;C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\Tools\;;C:\windows\system32;C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin;C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja
    SET PWD=/proc/self/cwd
    SET PYTHON_BIN_PATH=C:/Users/hbeck3/tf-gpu/env/Scripts/python.exe
    SET PYTHON_LIB_PATH=C:/Users/hbeck3/tf-gpu/env/lib/site-packages
    SET TEMP=C:\Users\hbeck3\AppData\Local\Temp
    SET TF_CUDA_CLANG=0
    SET TF_CUDA_COMPUTE_CAPABILITIES=5.0
    SET TF_CUDA_VERSION=10.0
    SET TF_CUDNN_VERSION=7
    SET TF_NEED_CUDA=1
    SET TF_NEED_OPENCL_SYCL=0
    SET TF_NEED_ROCM=0
    SET TMP=C:\Users\hbeck3\AppData\Local\Temp
  external/local_config_cuda/crosstool/windows/msvc_wrapper_for_nvcc.bat /nologo /DCOMPILER_MSVC /DNOMINMAX /D_WIN32_WINNT=0x0600 /D_CRT_SECURE_NO_DEPRECATE /D_CRT_SECURE_NO_WARNINGS /D_SILENCE_STDEXT_HASH_DEPRECATION_WARNINGS /bigobj /Zm500 /J /Gy /GF /EHsc /wd4351 /wd4291 /wd4250 /wd4996 /I. /Ibazel-out/x64_windows-opt/genfiles /Ibazel-out/x64_windows-opt/bin /Iexternal/com_google_absl /Ibazel-out/x64_windows-opt/genfiles/external/com_google_absl /Ibazel-out/x64_windows-opt/bin/external/com_google_absl /Iexternal/bazel_tools /Ibazel-out/x64_windows-opt/genfiles/external/bazel_tools /Ibazel-out/x64_windows-opt/bin/external/bazel_tools /Iexternal/eigen_archive /Ibazel-out/x64_windows-opt/genfiles/external/eigen_archive /Ibazel-out/x64_windows-opt/bin/external/eigen_archive /Iexternal/local_config_sycl /Ibazel-out/x64_windows-opt/genfiles/external/local_config_sycl /Ibazel-out/x64_windows-opt/bin/external/local_config_sycl /Iexternal/nsync /Ibazel-out/x64_windows-opt/genfiles/external/nsync /Ibazel-out/x64_windows-opt/bin/external/nsync /Iexternal/gif_archive /Ibazel-out/x64_windows-opt/genfiles/external/gif_archive /Ibazel-out/x64_windows-opt/bin/external/gif_archive /Iexternal/jpeg /Ibazel-out/x64_windows-opt/genfiles/external/jpeg /Ibazel-out/x64_windows-opt/bin/external/jpeg /Iexternal/protobuf_archive /Ibazel-out/x64_windows-opt/genfiles/external/protobuf_archive /Ibazel-out/x64_windows-opt/bin/external/protobuf_archive /Iexternal/com_googlesource_code_re2 /Ibazel-out/x64_windows-opt/genfiles/external/com_googlesource_code_re2 /Ibazel-out/x64_windows-opt/bin/external/com_googlesource_code_re2 /Iexternal/farmhash_archive /Ibazel-out/x64_windows-opt/genfiles/external/farmhash_archive /Ibazel-out/x64_windows-opt/bin/external/farmhash_archive /Iexternal/fft2d /Ibazel-out/x64_windows-opt/genfiles/external/fft2d /Ibazel-out/x64_windows-opt/bin/external/fft2d /Iexternal/highwayhash /Ibazel-out/x64_windows-opt/genfiles/external/highwayhash /Ibazel-out/x64_windows-opt/bin/external/highwayhash /Iexternal/zlib_archive /Ibazel-out/x64_windows-opt/genfiles/external/zlib_archive /Ibazel-out/x64_windows-opt/bin/external/zlib_archive /Iexternal/double_conversion /Ibazel-out/x64_windows-opt/genfiles/external/double_conversion /Ibazel-out/x64_windows-opt/bin/external/double_conversion /Iexternal/snappy /Ibazel-out/x64_windows-opt/genfiles/external/snappy /Ibazel-out/x64_windows-opt/bin/external/snappy /Iexternal/local_config_cuda /Ibazel-out/x64_windows-opt/genfiles/external/local_config_cuda /Ibazel-out/x64_windows-opt/bin/external/local_config_cuda /Iexternal/eigen_archive /Ibazel-out/x64_windows-opt/genfiles/external/eigen_archive /Ibazel-out/x64_windows-opt/bin/external/eigen_archive /Iexternal/nsync/public /Ibazel-out/x64_windows-opt/genfiles/external/nsync/public /Ibazel-out/x64_windows-opt/bin/external/nsync/public /Iexternal/gif_archive/lib /Ibazel-out/x64_windows-opt/genfiles/external/gif_archive/lib /Ibazel-out/x64_windows-opt/bin/external/gif_archive/lib /Iexternal/gif_archive/windows /Ibazel-out/x64_windows-opt/genfiles/external/gif_archive/windows /Ibazel-out/x64_windows-opt/bin/external/gif_archive/windows /Iexternal/protobuf_archive/src /Ibazel-out/x64_windows-opt/genfiles/external/protobuf_archive/src /Ibazel-out/x64_windows-opt/bin/external/protobuf_archive/src /Iexternal/farmhash_archive/src /Ibazel-out/x64_windows-opt/genfiles/external/farmhash_archive/src /Ibazel-out/x64_windows-opt/bin/external/farmhash_archive/src /Iexternal/zlib_archive /Ibazel-out/x64_windows-opt/genfiles/external/zlib_archive /Ibazel-out/x64_windows-opt/bin/external/zlib_archive /Iexternal/double_conversion /Ibazel-out/x64_windows-opt/genfiles/external/double_conversion /Ibazel-out/x64_windows-opt/bin/external/double_conversion /Iexternal/local_config_cuda/cuda /Ibazel-out/x64_windows-opt/genfiles/external/local_config_cuda/cuda /Ibazel-out/x64_windows-opt/bin/external/local_config_cuda/cuda /Iexternal/local_config_cuda/cuda/cuda/include /Ibazel-out/x64_windows-opt/genfiles/external/local_config_cuda/cuda/cuda/include /Ibazel-out/x64_windows-opt/bin/external/local_config_cuda/cuda/cuda/include /Iexternal/local_config_cuda/cuda/cuda/include/crt /Ibazel-out/x64_windows-opt/genfiles/external/local_config_cuda/cuda/cuda/include/crt /Ibazel-out/x64_windows-opt/bin/external/local_config_cuda/cuda/cuda/include/crt /D__CLANG_SUPPORT_DYN_ANNOTATION__ /DEIGEN_MPL2_ONLY /DEIGEN_MAX_ALIGN_BYTES=64 /DTF_USE_SNAPPY /showIncludes /MD /O2 /DNDEBUG -w /arch:AVX2 -DGOOGLE_CUDA=1 -DTENSORFLOW_MONOLITHIC_BUILD /DPLATFORM_WINDOWS /DEIGEN_HAS_C99_MATH /DTENSORFLOW_USE_EIGEN_THREADPOOL /DEIGEN_AVOID_STL_ARRAY /Iexternal/gemmlowp /wd4018 /wd4577 /DNOGDI /DTF_COMPILE_LIBRARY -DGOOGLE_CUDA=1 /Fobazel-out/x64_windows-opt/bin/tensorflow/core/kernels/data/_objs/parse_example_dataset_op/parse_example_dataset_op.o /c tensorflow/core/kernels/data/parse_example_dataset_op.cc
C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.15.26726\include\xmemory(217): error C2678: binary '*': no operator found which takes a left-hand operand of type 'const _Iter' (or there is no acceptable conversion)
        with
        [
            _Iter=tensorflow::OpInputList::Iterator
        ]
.\tensorflow/core/framework/op_kernel.h(405): note: could be 'const tensorflow::Tensor &tensorflow::OpArgIterator<tensorflow::OpInputList,const tensorflow::Tensor>::operator *(void)'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.15.26726\include\xmemory(217): note: while trying to match the argument list '(const _Iter)'
        with
        [
            _Iter=tensorflow::OpInputList::Iterator
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.15.26726\include\vector(1823): note: see reference to function template instantiation '_FwdIt *std::_Uninitialized_copy<_Iter,tensorflow::Tensor*,std::allocator<_Ty>>(const _InIt,const _InIt,_FwdIt,_Alloc &)' being compiled
        with
        [
            _FwdIt=tensorflow::Tensor *,
            _Iter=tensorflow::OpInputList::Iterator,
            _Ty=tensorflow::Tensor,
            _InIt=tensorflow::OpInputList::Iterator,
            _Alloc=std::allocator<tensorflow::Tensor>
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.15.26726\include\vector(708): note: see reference to function template instantiation 'tensorflow::Tensor *std::vector<tensorflow::Tensor,std::allocator<_Ty>>::_Ucopy<_Iter>(_Iter,_Iter,tensorflow::Tensor *)' being compiled
        with
        [
            _Ty=tensorflow::Tensor,
            _Iter=tensorflow::OpInputList::Iterator
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.15.26726\include\vector(708): note: see reference to function template instantiation 'tensorflow::Tensor *std::vector<tensorflow::Tensor,std::allocator<_Ty>>::_Ucopy<_Iter>(_Iter,_Iter,tensorflow::Tensor *)' being compiled
        with
        [
            _Ty=tensorflow::Tensor,
            _Iter=tensorflow::OpInputList::Iterator
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.15.26726\include\vector(723): note: see reference to function template instantiation 'void std::vector<tensorflow::Tensor,std::allocator<_Ty>>::_Range_construct_or_tidy<_Iter>(_Iter,_Iter,std::forward_iterator_tag)' being compiled
        with
        [
            _Ty=tensorflow::Tensor,
            _Iter=tensorflow::OpInputList::Iterator
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.15.26726\include\vector(723): note: see reference to function template instantiation 'void std::vector<tensorflow::Tensor,std::allocator<_Ty>>::_Range_construct_or_tidy<_Iter>(_Iter,_Iter,std::forward_iterator_tag)' being compiled
        with
        [
            _Ty=tensorflow::Tensor,
            _Iter=tensorflow::OpInputList::Iterator
        ]
tensorflow/core/kernels/data/parse_example_dataset_op.cc(91): note: see reference to function template instantiation 'std::vector<tensorflow::Tensor,std::allocator<_Ty>>::vector<tensorflow::OpInputList::Iterator,void>(_Iter,_Iter,const _Alloc &)' being compiled
        with
        [
            _Ty=tensorflow::Tensor,
            _Iter=tensorflow::OpInputList::Iterator,
            _Alloc=std::allocator<tensorflow::Tensor>
        ]
tensorflow/core/kernels/data/parse_example_dataset_op.cc(90): note: see reference to function template instantiation 'std::vector<tensorflow::Tensor,std::allocator<_Ty>>::vector<tensorflow::OpInputList::Iterator,void>(_Iter,_Iter,const _Alloc &)' being compiled
        with
        [
            _Ty=tensorflow::Tensor,
            _Iter=tensorflow::OpInputList::Iterator,
            _Alloc=std::allocator<tensorflow::Tensor>
        ]
external/com_google_absl\absl/container/inlined_vector.h(73): note: see reference to class template instantiation 'std::allocator<_Ty>' being compiled
        with
        [
            _Ty=Eigen::DenseIndex
        ]
.\tensorflow/core/util/tensor_format.h(451): note: see reference to class template instantiation 'absl::InlinedVector<tensorflow::int64,4,std::allocator<_Ty>>' being compiled
        with
        [
            _Ty=Eigen::DenseIndex
        ]
external/com_google_absl\absl/container/inlined_vector.h(73): note: see reference to class template instantiation 'std::allocator<T>' being compiled
        with
        [
            T=tensorflow::MemoryType
        ]
.\tensorflow/core/framework/op_kernel.h(177): note: see reference to class template instantiation 'absl::InlinedVector<tensorflow::MemoryType,4,std::allocator<T>>' being compiled
        with
        [
            T=tensorflow::MemoryType
        ]
external/com_google_absl\absl/container/inlined_vector.h(73): note: see reference to class template instantiation 'std::allocator<T>' being compiled
        with
        [
            T=tensorflow::TensorReference
        ]
.\tensorflow/core/framework/unique_tensor_references.h(66): note: see reference to class template instantiation 'absl::InlinedVector<tensorflow::TensorReference,4,std::allocator<T>>' being compiled
        with
        [
            T=tensorflow::TensorReference
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.15.26726\include\xmemory(217): error C2100: illegal indirection
C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.15.26726\include\xmemory(217): error C2062: type 'unknown-type' unexpected
Target //tensorflow/tools/pip_package:build_pip_package failed to build
INFO: Elapsed time: 636,778s, Critical Path: 251,57s
INFO: 783 processes: 783 local.
FAILED: Build did NOT complete successfully```

**Comment by yunghoy** [comment]
Thank you! It works with the solution.

**Comment by dtarakanov1** [comment]
Does the fix go to 1.12.1? (it's on the master)

**Comment by mika-fischer** [comment]
Still present in 1.13.0-rc2...
